packages/core/crates/core-p2p/src/lib.rs

```diff
@@ -1,15 +1,81 @@
+use futures::{select, StreamExt};
+use libp2p::StreamProtocol;
+
 pub use libp2p::identity;
 
 use libp2p::identity as libp2p_identity;
 use libp2p::core as libp2p_core;
 use libp2p::swarm as libp2p_swarm;
 use libp2p::noise as libp2p_noise;
+use libp2p::request_response as libp2p_request_response;
 
 use libp2p_identity::PeerId;
 use libp2p_core::{upgrade, Transport};
-use libp2p_swarm::{SwarmBuilder, SwarmEvent};
+use libp2p_swarm::{NetworkBehaviour, SwarmBuilder, SwarmEvent};
+
+use serde::{Serialize, Deserialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Ping(ControlMessage);
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Pong(ControlMessage);
+
+use core_network::messaging::ControlMessage;
+
+pub const HOPR_HEARTBEAT_PROTOCOL_V_0_1_0: &str = "/hopr/heartbeat/0.1.0";
+pub const HOPR_MESSAGE_PROTOCOL_V_0_1_0: &str = "/hopr/msg/0.1.0";
+pub const HOPR_ACKNOWLEDGE_PROTOCOL_V_0_1_0: &str = "/hopr/ack/0.1.0";
```

Comment:
Shouldn't we use version 2.0.0 ?

Reply:
This is more or less a placeholder. The protocol name and version number can change for the final PR merge.